subcommand/mesh-init/command.go

```diff
@@ -178,211 +212,6 @@ func (c *Command) realRun() error {
-		}
-
-		tok, _, err := client.ACL().Login(
-			c.constructLoginParams(bearerToken, taskMeta),
-			&api.WriteOptions{Partition: partition},
-		)
-		if err != nil {
-			c.log.Error(err.Error())
-			return err
-		}
-
-		err = os.WriteFile(tokenFile, []byte(tok.SecretID), 0644)
-		if err != nil {
-			return err
-		}
-
-		c.log.Info("login success", "accessor-id", tok.AccessorID, "token-file", tokenFile)
-		return nil
-	}, backoff.NewConstantBackOff(2*time.Second), retryLogger(c.log))
-}
-
-func (c *Command) constructLoginParams(bearerToken string, taskMeta awsutil.ECSTaskMeta) *api.ACLLoginParams {
-	method := c.config.ConsulLogin.Method
-	if method == "" {
-		method = config.DefaultAuthMethodName
-	}
-	meta := mergeMeta(
-		map[string]string{
-			"consul.hashicorp.com/task-id": taskMeta.TaskID(),
-			"consul.hashicorp.com/cluster": taskMeta.Cluster,
-		},
-		c.config.ConsulLogin.Meta,
-	)
-	return &api.ACLLoginParams{
-		AuthMethod:  method,
-		BearerToken: bearerToken,
-		Meta:        meta,
-	}
-}
-
-func (c *Command) createAWSBearerToken(taskMeta awsutil.ECSTaskMeta) (string, error) {
-	l := c.config.ConsulLogin
-
-	region := l.Region
-	if region == "" {
-		r, err := taskMeta.Region()
-		if err != nil {
-			return "", err
-		}
-		region = r
-	}
-
-	cfg := aws.Config{
-		Region: aws.String(region),
-		// More detailed error message to help debug credential discovery.
-		CredentialsChainVerboseErrors: aws.Bool(true),
-	}
-
-	// support explicit creds for unit tests
-	if l.AccessKeyID != "" {
-		cfg.Credentials = credentials.NewStaticCredentials(
-			l.AccessKeyID, l.SecretAccessKey, "",
-		)
-	}
-
-	// Session loads creds from standard sources (env vars, file, EC2 metadata, ...)
-	sess, err := session.NewSessionWithOptions(session.Options{
-		Config: cfg,
-		// Allow loading from config files by default:
-		//   ~/.aws/config or AWS_CONFIG_FILE
-		//   ~/.aws/credentials or AWS_SHARED_CREDENTIALS_FILE
-		SharedConfigState: session.SharedConfigEnable,
-	})
-	if err != nil {
-		return "", err
-	}
-
-	if sess.Config.Credentials == nil {
-		return "", fmt.Errorf("AWS credentials not found")
-	}
-
-	loginData, err := iamauth.GenerateLoginData(&iamauth.LoginInput{
-		Creds:                  sess.Config.Credentials,
-		IncludeIAMEntity:       l.IncludeEntity,
-		STSEndpoint:            l.STSEndpoint,
-		STSRegion:              region,
-		Logger:                 hclog.New(nil),
-		ServerIDHeaderValue:    l.ServerIDHeaderValue,
-		ServerIDHeaderName:     config.IAMServerIDHeaderName,
-		GetEntityMethodHeader:  config.GetEntityMethodHeader,
-		GetEntityURLHeader:     config.GetEntityURLHeader,
-		GetEntityHeadersHeader: config.GetEntityHeadersHeader,
-		GetEntityBodyHeader:    config.GetEntityBodyHeader,
-	})
-	if err != nil {
-		return "", err
-	}
-
-	loginDataJson, err := json.Marshal(loginData)
-	if err != nil {
-		return "", err
-	}
-	return string(loginDataJson), err
-}
-
-func (c *Command) waitForTokenReplication(tokenFile string) error {
```

Comment:
Note: I think it's fine to remove this for now, because of the [sleep hack](https://github.com/hashicorp/consul-server-connection-manager/blob/main/discovery/acl.go#L68-L73) in the connection manager lib. It may be worth adding a comment about this where the ACL login now happens, in case that sleep hack isn't enough to reliably prevent additional cases of this issue (which it likely isn't).